lua/gatekeeper/staticlib.lua

```diff
@@ -124,6 +124,38 @@ function convert_numa_table_to_array (numa_table)
 	return res
 end
 
+function gk_sol_map (gk_lcores, sol_lcores)
+	local m = {}
+	local d = #gk_lcores / #sol_lcores
+	local sol_remaining_cap = {}
+
+	if #gk_lcores % #sol_lcores ~= 0 then
+		error("Wrong GK-to-SOL assignment");
+	end
```

Comment:
Shouldn't it just be a warning? Suppose that one needs 3 GK blocks on a two NUMA node machine. Since we'are going to allocate 2 SOL blocks, this test fails. But I think that the setup, although unbalanced, should be valid.